bubble_sort_by.rb

```diff
@@ -0,0 +1,15 @@
+def bubble_sort_by(arr)
+  loop do
+    sorted = false
+    (0...arr.length - 1).each do |i|
+      if (arr[i].length - arr[i + 1].length).positive?
+        arr[i], arr[i + 1] = arr[i + 1], arr[i]
+        sorted = true
+      end
+    end
+    break unless sorted
+  end
+  arr
+end
```

Comment:
Project Specification (**requirements from Odin**)

* [ ] bubble_sort_by method is implemented and working ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_bubble_sort/sortBy.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-1))

- bubble_sort_by has to use yield and only works if a block is given if not it should retrieve an error your's returns the unsorted array when no block is given

![image](https://user-images.githubusercontent.com/52209067/78667463-2f2b0f00-7896-11ea-8768-c4698a356191.png)

- also, when trying to test with an array of integers is retrieving a NoMethodError so let's fix it!:
![image](https://user-images.githubusercontent.com/52209067/78667955-15d69280-7897-11ea-9248-96153c150558.png)